### PR TITLE
Helpers for reporting controller errors as K8s events

### LINKED ISSUE #321: K8s error logging
Compared to #315 this only covers logging fatal reconciliation events. Full logging will come later...

### PR DESCRIPTION
**Pull request by teozkr**
## Description

Fixes #321

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)

### CODE REVIEW

File: src/logging/controller.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    /// A reference to a secondary object providing additional context, if any
+    ///
+    /// This should be [`Some`] if the error happens while evaluating some related object
+    /// (for example: when writing a [`StatefulSet`] owned by your controller object).
+    fn secondary_object(&self) -> Option<ObjectRef<DynamicObject>> {
+        None
+    }
+}
+
+/// Reports the controller reconciliation result to all relevant targets
```
Comment (maltesander) [inline src/logging/controller.rs:37]: /// Reports the controller reconciliation result to all relevant targets:
/// * Console for the operator administrator
/// * Kubernetes event (if relevant) to the end user

**Review by maltesander — APPROVED** [review]
LGTM.

File: src/logging/controller.rs (hunk 1 above)
Comment (teozkr) [reply src/logging/controller.rs:37]: Rephrased a bit and committed as 0798470

**Pull request merged**